metadata-ingestion/src/datahub/sql_parsing/sql_parsing_aggregator.py

```diff
@@ -905,9 +924,43 @@ def add_table_rename(
 
         self.report.num_table_renames += 1
 
+        if original_urn in self._lineage_map:
+            self._lineage_map[new_urn] = self._lineage_map.pop(original_urn)
+
         # This will not work if the table is renamed multiple times.
         self._table_renames[original_urn] = new_urn
 
+    def add_table_swap(
+        self,
+        urn1: UrnStr,
+        urn2: UrnStr,
+    ) -> None:
+        """Add a table swap to the aggregator.
+
+        Args:
+            urn1, urn2: The dataset URNs to swap.
+        """
+
+        self.report.num_table_swaps += 1
+
+        urn1_lineage: Optional[OrderedSet[QueryId]] = None
+        urn2_lineage: Optional[OrderedSet[QueryId]] = None
+
+        if urn1 in self._lineage_map:
+            urn1_lineage = self._lineage_map.pop(urn1)
+
+        if urn2 in self._lineage_map:
+            urn2_lineage = self._lineage_map.pop(urn2)
+
+        if urn1_lineage:
+            self._lineage_map[urn2] = urn1_lineage
+
+        if urn2_lineage:
+            self._lineage_map[urn1] = urn2_lineage
+
+        self._table_renames[urn1] = urn2
+        self._table_renames[urn2] = urn1
```

Comment:
if we see a swap statement multiple times, will we get messed up by that?

Reply:
Yes, good catch. We can deduplicate swaps or process only once. That should solve this.